Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.cs

```diff
@@ -254,18 +254,27 @@ public void Decompose(IdentityAddress activeIdentity, DeviceId activeDevice)
     {
         EnsureHasParticipant(activeIdentity);
         EnsureRelationshipNotDecomposedBy(activeIdentity);
-        EnsureStatus(RelationshipStatus.Terminated);
 
         if (From == activeIdentity)
             FromHasDecomposed = true;
 
         if (To == activeIdentity)
             ToHasDecomposed = true;
```

Comment:
It's weird that here part of the decomposition is already made, and down below there methods with the names DecomposeAs[First/Second]Participant are called.

Reply:
Yes it is, I should have given it another thought.
It's refactored now.